go/marshal/decode.go

```diff
@@ -217,3 +238,51 @@ func nomsValueDecoder(v types.Value, rv reflect.Value) {
 	}
 	rv.Set(reflect.ValueOf(v))
 }
+
+func sliceDecoder(t reflect.Type) decoderFunc {
+	d := decoderCache.get(t)
+	if d != nil {
+		return d
+	}
+
+	var decoder decoderFunc
+
+	d = func(v types.Value, rv reflect.Value) {
+		slice := rv.Slice(0, 0)
```

Comment:
You should be able to pre-allocate this slice to be the right length, right? To avoid having to grow it as items are appended.

Also, I wonder if it's worth seeing if rv is already a slice that has enough capacity for the list contents?